ios/Capacitor/Capacitor/Plugins/CapacitorUrlRequest.swift

```diff
@@ -92,8 +92,67 @@ public class CapacitorUrlRequest: NSObject, URLSessionTaskDelegate {
 
         return normalized[index.lowercased()]
     }
+    
+    func getRequestDataFromFormData(_ data: JSValue) throws -> Data? {
+        guard let list = data as? JSArray else {
+            // Throw, other data types explicitly not supported.
+            throw CapacitorUrlRequestError.serializationError("Data must be an array for FormData")
+        }
+        
+        var data = Data()
+        
+        // Update the contentType with the new boundary
+        let boundary = UUID().uuidString
+        let contentType = "multipart/form-data; boundary=\(boundary)"
+        request.setValue(contentType, forHTTPHeaderField: "Content-Type")
+        headers["Content-Type"] = contentType
+        
+        for entry in list {
+            guard let item = entry as? [String: String] else {
+                throw CapacitorUrlRequestError.serializationError("Data must be an array for FormData")
+            }
+            
+            let type = item["type"]
+            let key = item["key"]
+            let value = item["value"]!
+            
+            if type == "base64File" {
+                let fileName = item["fileName"]
+                let fileContentType = item["contentType"]
+                
+                data.append("\r\n--\(boundary)\r\n".data(using: .utf8)!)
+                data.append("Content-Disposition: form-data; name=\"\(key!)\"; filename=\"\(fileName!)\"\r\n".data(using: .utf8)!)
+                data.append("Content-Type: \(fileContentType!)\r\n".data(using: .utf8)!)
+                data.append("\r\n".data(using: .utf8)!)
+                
+                data.append(Data(base64Encoded: value)!)
+                
+                data.append("\r\n".data(using: .utf8)!)
+            } else if type == "string" {
+                data.append("\r\n--\(boundary)\r\n".data(using: .utf8)!)
+                data.append("Content-Disposition: form-data; name=\"\(key!)\"\r\n".data(using: .utf8)!)
+                data.append(value.data(using: .utf8)!)
+                data.append("\r\n".data(using: .utf8)!)
+            }
+            
+        }
+        
+        
+        data.append("\r\n--\(boundary)--\r\n".data(using: .utf8)!)
+        
+        return data;
+    }
 
-    func getRequestData(_ body: JSValue, _ contentType: String) throws -> Data? {
+    func getRequestData(_ body: JSValue, _ contentType: String, _ dataType: String) throws -> Data? {
+        if dataType == "file" {
+            guard let stringData = body as? String else {
+                throw CapacitorUrlRequestError.serializationError("[ data ] argument could not be parsed as string")
+            }
+            return Data(base64Encoded: stringData)
+        } else if dataType == "formData" {
```

Comment:
There is existing code for handling `multipart/form-data` based on the contentType, but it doesn't seem to allow `FormData` objects, just plain js objects. It seems a bit weird as there is no Android code for this, and I don't think fetch would support it either? 

So I added it as an extra case so as not to break existing functionality